src/python/restapi/RestAPI.py

```diff
@@ -13,34 +13,48 @@
 from fastapi import APIRouter, Depends, FastAPI, HTTPException, Request, Response, status
 from fastapi.security import HTTPBearer, HTTPBasic
 
-#from fastapi.responses import JSONResponse
+from fastapi.responses import JSONResponse
 from typing import Dict, List
 
 from pdmodels.Models import DeviceNote, PhysicalDevice, LogicalDevice, PhysicalToLogicalMapping
 import api.client.DAO as dao
 
 import base64
 
+# Prometheus metrics
+from prometheus_client import Counter, Histogram, Gauge, start_http_server
+from prometheus_fastapi_instrumentator import Instrumentator
+request_counter = Counter('requests_total', 'Total Requests')
+errors_counter = Counter('errors_total', 'Total Errors')
+
+# Start up the server to expose the metrics.
+start_http_server(8000)
+
 # Scheme for the Authorization header
 token_auth_scheme = HTTPBearer()
 http_basic_auth = HTTPBasic()
 
 # router = APIRouter(prefix='/broker/api', dependencies=[Depends(token_auth_scheme)])
 router = APIRouter(prefix='/broker/api')
 
+
 @router.get("/physical/sources/", tags=['physical devices'], dependencies=[Depends(token_auth_scheme)])
 async def get_all_physical_sources() -> List[str]:
+    request_counter.inc()
     """
     Return a list of all physical device sources.
     """
     try:
         return dao.get_all_physical_sources()
     except dao.DAOException as err:
+        errors_counter.inc()
         raise HTTPException(status_code=500, detail=err.msg)
 
 
 @router.get("/physical/devices/", tags=['physical devices'], dependencies=[Depends(token_auth_scheme)])
-async def query_physical_devices(source_name: str = None, include_properties: bool | None = True) -> List[PhysicalDevice]:
+async def query_physical_devices(source_name: str = None, include_properties: bool | None = True) -> List[
```

Comment:
You seem to have changed some styling in this file.
There are a few changes which are style only.

Reply:
Ah yes, my IDE does code formatting/spell checking. I'll revert style back to original forms.